[OSDOCS#18857]: Added a new module for Fencing Validator script (#110148)
openshift/openshift-docs: modules/Installation-fencing-validator-disruptive-tests.adoc

:_mod-docs-content-type: CONCEPT
[id="installation-fencing-validator-disruptive-checks_{context}"]
= Run fencing validator script for disruptive checks

[role="_abstract"]

> 

Reviewer: Remove the blank line between the abstract tag and the abstract sentence.

> For disruptive checks, the `fencing_validator` script runs the following command from the peer node: 

Reviewer: The abstract should not be a lead-in sentence. The sentence should state the what and why.Something like the following but it would need SME and QE validation: 

> To validate your cluster's resilience, use the fencing_validator script to perform disruptive checks from a peer node. Executing these simulated failures helps you ensure your high-availability environment correctly isolates and recovers from errors.

> [source,bash]

Reviewer: The script would run this command in the terminal so [source,terminal].

Reviewer: <img width="916" height="202" alt="Image" src="https://github.com/user-attachments/assets/cdbcdf42-b790-47c0-a87e-aa8dc3ff851d" />

> pcs stonith fence <node>

Reviewer: ```suggestion
[source,bash]
----
pcs stonith fence <node>
----
```

A lot of these code blocks are missing the delimiters.

You can check whether Two-Node with Fencing (TNF) cluster setup actually works by running the following command: 

 [source,bash]

> 

Reviewer: No harm putting the dollar sign before the command to represent a regular user. See https://redhat-documentation.github.io/supplementary-style-guide/#commands-with-root-privileges

oc debug node/\<node-name\> -- chroot /host /usr/local/bin/fencing\_validator --disruptive

[WARNING]

> 

Reviewer: The [WARNING] tag should be above the delimiter (====). No blank line should go between them.

====

> The `--disruptive` flag fences each control-plane node one at a time and verifies recovery. It causes temporary workload disruption. Only use this during maintenance windows or initial validation. This is the STONITH fence operation such as power cycle or VM reset, and not a graceful shutdown.

Reviewer: ```suggestion
The `--disruptive` flag fences each control-plane node one at a time and verifies recovery. The flag causes temporary workload disruption. Use the flag only during maintenance windows or initial validation. This is the STONITH fence operation such as power cycle or VM reset, and not a graceful shutdown.
```

I'm not sure what "This is the STONITH fence operation". Does this statement relate to what the commands in this procedure does and the commands are not related to graceful shutdown?

====

The fencing validator script with `--disruptive` flag runs the following checks: 

. **Fence Node A** - Triggers STONITH to reboot the first control-plane node.

> . **Verify NotReady** - Waits for Kubernetes to report the node as `NotReady` confirming the reboot happened.

Reviewer: ```suggestion
. **Verify NotReady** - Waits for Kubernetes to report the node as `NotReady`, which confirms the reboot happened.
```

> . **Verify recovery** - Waits for the node to come back `Ready`, rejoin the Pacemaker cluster as `ONLINE`, and for etcd to regain quorum.

Reviewer: ```suggestion
. **Verify recovery** - Waits for the node to come back to the `Ready` state, rejoin the Pacemaker cluster as `ONLINE`, and for etcd to regain quorum.
```

:_mod-docs-content-type: CONCEPT
[id="installation-fencing-validator-disruptive-checks_{context}"]

> = Run fencing validator script for disruptive checks

Reviewer: Considering turning this file into a procedure as verbs should not exist in a concept title.

[role="_abstract"]

> You can validate your cluster's resilience and perform disruptive checks from a peer node by using the `fencing_validator` script. Executing these simulated failures helps you ensure your high-availability environment correctly isolates and recovers from errors.

Reviewer: ```suggestion
You can validate your cluster's resilience and perform disruptive checks from a peer node by using the `fencing_validator` script. By executing these simulated failures, you can ensure your high-availability environment correctly isolates and recovers from errors.
```

Removes anthorphomorphism

[source,terminal]
----
$ pcs stonith fence <node>
----

You can check whether Two-Node with Fencing (TNF) cluster setup actually works by running the following command: 

[source,terminal]
----
$ oc debug node/<node-name> --chroot /host /usr/local/bin/fencing_validator --disruptive
----

[WARNING]
====

> The `--disruptive` flag fences each control-plane node one at a time and verifies recovery. It causes temporary workload disruption. Only use this during maintenance windows or initial validation. The `--disruptive` flag actually performs the Shoot The Other Node In The Head(STONITH) fence operations such as power cycle or VM reset, and not a graceful shutdown.

Reviewer: ```suggestion
The `--disruptive` flag fences each control-plane node one at a time and verifies recovery. The flag causes temporary workload disruption. Only use this during maintenance windows or initial validation. The `--disruptive` flag actually performs the Shoot The Other Node In The Head (STONITH) fence operations such as power cycle or VM reset, and not a graceful shutdown.
```
